src/util/mw-api.ts

```diff
@@ -114,7 +114,7 @@ export function normalizeMwResponse(response: MwApiQueryResponse): QueryMwRet {
 
     return Object.values(pages)
         .reduce((acc, page) => {
-            const id = (normalized[page.title] || page.title || '');
+            const id = ((normalized.hasOwnProperty(page.title) && normalized[page.title]) || page.title || '');
```

Comment:
I don't understand exactly why this is needed and how this fixes the problem.

Reply:
@kelson42 `normalize` doesn't have any element named constructor so by default all arrays are also special kind of objects inJS so when we write `normalized[page.title]` it tries to access the default constructor inherited from object class and assign that to `id`, using `normalized.hasOwnProperty(page.title)` checks if it is it's own property and not inherited property.

Reply:
Why `normalized` does not have an element at the key `constructor`? It should, this is how the code works! Why we have a problem now considering that this code has not been changed?

Reply:
@kelson42 `normalized` is not present in case of Wiktionary as you can see [here](https://en.wiktionary.org/w/api.php?action=query&format=json&prop=redirects%7Crevisions%7Cpageimages&rdlimit=max&rdnamespace=0&redirects=true&titles=constructor)

Reply:
ok, lets try this.